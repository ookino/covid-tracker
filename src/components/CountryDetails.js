/* eslint-disable react/prop-types */
import React from 'react';

const CountryDetails = ({ country }) => (
  <ul className="details__ul">
    <li className="details__li">
      <p>Continent</p>
      <p>{country.continent}</p>
    </li>
    <li className="details__li">
      <p>Population</p>
      <p>{country.population}</p>
    </li>
    <li className="details__li">
      <p>Cases</p>
      <p>{country.cases}</p>
    </li>
    <li className="details__li">
      <p>Critical Cases</p>
      <p>{country.critical}</p>
    </li>
    <li className="details__li">
      <p>Recovered</p>
      <p>{country.recovered}</p>
    </li>
    <li className="details__li">
      <p>Deaths</p>
      <p>{country.deaths}</p>
    </li>
  </ul>
);
export default CountryDetails;
