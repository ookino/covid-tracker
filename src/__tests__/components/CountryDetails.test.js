import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import CountryDetails from '../../components/CountryDetails';

it('renders CountryDetails Component correctly', () => {
  const country = {
    continent: 'Africa',
    population: 523464,
    cases: 464574,
    critical: 32453,
    recovered: 423534,
    deaths: 2453,
  };
  const tree = renderer
    .create(
      <Router>
        <CountryDetails country={country} />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
