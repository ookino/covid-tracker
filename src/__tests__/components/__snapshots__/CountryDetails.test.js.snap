// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CountryDetails Component correctly 1`] = `
<ul
  className="details__ul"
>
  <li
    className="details__li"
  >
    <p>
      Continent
    </p>
    <p>
      Africa
    </p>
  </li>
  <li
    className="details__li"
  >
    <p>
      Population
    </p>
    <p>
      523464
    </p>
  </li>
  <li
    className="details__li"
  >
    <p>
      Cases
    </p>
    <p>
      464574
    </p>
  </li>
  <li
    className="details__li"
  >
    <p>
      Critical Cases
    </p>
    <p>
      32453
    </p>
  </li>
  <li
    className="details__li"
  >
    <p>
      Recovered
    </p>
    <p>
      423534
    </p>
  </li>
  <li
    className="details__li"
  >
    <p>
      Deaths
    </p>
    <p>
      2453
    </p>
  </li>
</ul>
`;
